x/retry/config.go

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package retry
+
+import (
+	"fmt"
+	"time"
+
+	"go.uber.org/multierr"
+	iconfig "go.uber.org/yarpc/internal/config"
+	"go.uber.org/yarpc/x/config"
+)
+
+// PolicySpec defines how to construct a retry Policy.
+type PolicySpec struct {
```

Comment:
Is this Policy spec or Policy config? We've been using `*Spec` for things that specify how to construct more complex objects. We register the specs against the Configurator and then decode configurations, using the spec to build complex objects.

For example, for peer choosers, we register peer list specs, peer list updater specs, and we decode `config.PeerChooser` structs.